src/WebJobs.Extensions.DurableTask/ContextImplementations/DurableClient.cs

```diff
@@ -910,12 +910,27 @@ private static async Task<DurableOrchestrationStatus> GetDurableOrchestrationSta
                         }
                     }
 
-                    var counter = 0;
-                    indexList.Sort();
-                    foreach (var indexValue in indexList)
+                    if (eventTypes != null)
                     {
-                        historyArray.RemoveAt(indexValue - counter);
-                        counter++;
+                        for (int i = historyArray.Count - 1; i >= 0; i--)
+                        {
+                            var eventTypeName = historyArray[i].Value<string>("EventType");
+                            var eventType = (EventType)Enum.Parse(typeof(EventType), eventTypeName, true);
```

Comment:
The `.Parse` operation may fail, and I think it would be good to handle that failure. As a result, instead of calling `.Parse` directly, I recommend calling [`.TryParse`](https://learn.microsoft.com/en-us/dotnet/api/system.enum.tryparse?view=net-7.0) which is a bit safer and returns `true` if the parsing succeeded, and `false` if it failed. You can use that result to log a warning and abort the operation if the parsing fails.